src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -256,6 +259,53 @@ public ModalFragment(IMauiContext mauiContext, Page modal)
 				return dialog;
 			}
 
+			void OnPageHandlerChanged(object? sender, EventArgs e)
+			{
+				if (sender is Page page)
+				{
+					page.HandlerChanged -= OnPageHandlerChanged;
+				}
+
+				UpdateBackgroundColor();
+			}
+
+			void OnModalPagePropertyChanged(object? sender, PropertyChangedEventArgs e)
+			{
+				if (_modal is null)
+				{
+					if (sender is Page page)
+					{
+						page.PropertyChanged -= OnModalPagePropertyChanged;
+						page.HandlerChanged -= OnPageHandlerChanged;
+					}
+
+					return;
+				}
+
+
+				if (e.IsOneOf(Page.BackgroundColorProperty, Page.BackgroundProperty))
```

Comment:
I think BackgroundColor will always fire Background, so this is allocating an array twice and then firing twice for each time the Color is set.

Reply:
the `Background` Mapper will fire but the `Background` INPC won't fire

Here's a quick test
![image](https://github.com/user-attachments/assets/0fbef243-185c-4d1a-a147-a7b35a82387e)